lib/CMakeLists.txt

```diff
@@ -286,6 +286,23 @@ add_library(syslog-ng SHARED ${LIB_SOURCES})
 
 target_include_directories(syslog-ng SYSTEM INTERFACE ${CORE_INCLUDE_DIRS})
 
+function (openssl_set_define_for_symbol target symbol)
+  set (CMAKE_REQUIRED_INCLUDES ${OPENSSL_INCLUDE_DIR})
+  set (CMAKE_REQUIRED_LIBRARIES ${OPENSSL_LIBRARIES})
+  include(CheckSymbolExists)
+  check_symbol_exists(${symbol} "openssl/evp.h;openssl/ssl.h;openssl/x509v3.h" HAVE_${symbol})
+  if (HAVE_${symbol})
+    string(TOUPPER SYSLOG_NG_HAVE_DECL_${symbol} SYMBOL_WITH_CAPITALS)
+    target_compile_definitions(syslog-ng PUBLIC ${SYMBOL_WITH_CAPITALS}=1)
```

Comment:
The `SYSLOG_NG_HAVE...` compiler switch variables are provided via a header file `syslog-ng-config.in.h`. There is no need to directly set it with `target_compile_definitions`.